src/browser/modules/D3Visualization/lib/visualization/components/Visualization.ts

```diff
@@ -235,129 +132,115 @@ const vizFn = function (
-      const tolerance = 25
-
-      const dist = (a: [number, number], b: [number, number]) =>
-        Math.pow(a[0] - b[0], 2) + Math.pow(a[1] - b[1], 2)
-
-      function dragstarted(event: D3DragEvent<SVGGElement, VizNode, any>) {
-        initialPos = [event.x, event.y]
-        restartedSim = false
-      }
-
-      function dragged(
-        event: D3DragEvent<SVGGElement, VizNode, any>,
-        node: VizNode
-      ) {
-        // This is to prevent clicks/double clicks from restarting the simulation
-        if (dist(initialPos, [event.x, event.y]) > tolerance && !restartedSim) {
-          // Set alphaTarget to a value higher than alphaMin so the simulation
-          // isn't stopped while nodes are being dragged.
-          simulation.alphaTarget(0.09).alpha(1).restart()
-          restartedSim = true
-        }
+    this.forceSim.updateRelationships(this.graph)
+  }
 
-        node.hoverFixed = false
-        node.fx = event.x
-        node.fy = event.y
-      }
+  private zoomClick(isZoomingIn: any) {
+    this.draw = true
+    this.isZoomClick = true
+    const limitsReached = { zoomInLimit: false, zoomOutLimit: false }
 
-      function dragended(_event: D3DragEvent<SVGGElement, VizNode, any>) {
-        if (restartedSim) {
-          // Reset alphaTarget so the simulation cools down and stops.
-          simulation.alphaTarget(0)
-        }
-      }
+    this.zoomBehavior.scaleBy(this.baseGroup, isZoomingIn ? 1.3 : 0.7)
 
-      return d3Drag<SVGGElement, VizNode>()
-        .on('start', dragstarted)
-        .on('drag', dragged)
-        .on('end', dragended)
-    }
+    return limitsReached
+  }
 
-    const nodeGroups = container
-      .select('g.layer.nodes')
-      .selectAll<BaseType, VizNode>('g.node')
-      .data(nodes, d => d.id)
+  init() {
+    this.container
+      .selectAll('g.layer')
+      .data(['relationships', 'nodes'])
       .join('g')
-      .attr('class', 'node')
-      // @ts-ignore
-      .call(dragHandler(force.simulation))
-      .on('mouseover', onNodeMouseOver)
-      .on('mouseout', onNodeMouseOut)
-      .on('click', onNodeClick)
-      .on('dblclick', onNodeDblClick)
-      .classed('selected', node => node.selected)
+      .attr('class', d => `layer ${d}`)
 
-    for (const renderer of vizRenderers.node) {
-      nodeGroups.call(renderer.onGraphChange, viz)
-    }
+    this.updateNodes()
+    this.updateRelationships()
+    this.forceSim.precompute()
+  }
 
-    for (const renderer of nodeMenuRenderer) {
-      nodeGroups.call(renderer.onGraphChange, viz)
+  update(options: {
+    updateNodes: boolean
+    updateRelationships: boolean
+    rerender: boolean
+  }) {
+    if (options.updateNodes) {
+      this.updateNodes()
     }
 
-    if (updateViz) {
-      force.update(graph, options.precompute)
+    if (options.updateRelationships) {
+      this.updateRelationships()
+    }
 
-      viz.resize()
-      viz.trigger('updated')
+    if (options.rerender) {
+      this.forceSim.restart()
```

Comment:
What's the reason you call the option rerender rather than restartSimulation or similar?